content/_code-samples/secure-signing/py/sign-payment.py

```diff
@@ -0,0 +1,34 @@
+from xrpl.core import keypairs 
+
+# Define network client
+from xrpl.clients import JsonRpcClient
+JSON_RPC_URL = "https://s.altnet.rippletest.net:51234/"
+client = JsonRpcClient(JSON_RPC_URL)
+
+# Define signer address
+from xrpl.wallet import Wallet
+test_wallet_1 = Wallet(seed="shrPSF6vV3v3yoND9J3NeKks6M3xd")
```

Comment:
Leaving the secret hardcoded in the file like this is not actually that secure. The example JavaScript code pulls the secret from environment variables, which is a common technique for managing secrets of scripts, especially when you need to use them from fully-automated scripts/servers/bots.

Here's that JavaScript example:

```js
const from_address = process.env['MY_ADDRESS']
const secret = process.env['MY_SECRET']
```

The Python equivalent would be:

```py
import os
my_secret = os.getenv("MY_SECRET")
wallet = Wallet(seed=my_secret)
```

(Fun fact: there's a ([real example of this in txsplainbot](https://github.com/mDuo13/txsplain/blob/7b88660199970aa61e27256b3ae2f078d1f0c1d5/bot.py#L7).)

Of course, the environment variable thing is only as secure as whatever thing actually sets the environment variable before running the script. If you put `export MY_SECRET=shrPSF6vV3v3yoND9J3NeKks6M3xd` in your `.bashrc` or equivalent file, it's not a whole lot more secure than having it hard-coded (but it _is_ more flexible since you could run the same code with a different secret).

Another way to do it in Python would be to ask the user to type in a secret key using [`getpass()`](https://docs.python.org/3/library/getpass.html)—that way, they could have the key saved encrypted in a password manager and only paste it into the terminal when needed. It's not useful for automated systems, but it's a good basic-level way of getting the secret from the user that can stand-in for more specific/complicated equivalents that might come from more advanced (read: graphical) user interfaces.